plugins/node/opentelemetry-instrumentation-pulsar-client/src/instrumentation.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  InstrumentationBase,
+  InstrumentationNodeModuleDefinition,
+  isWrapped,
+} from '@opentelemetry/instrumentation';
+import type * as Pulsar from 'pulsar-client';
+import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
+import { InstrumentationConfig } from './types';
+import { VERSION } from './version';
+import { ClientProxy } from './proxies/client';
+
+type PulsarConstructor = new (config: Pulsar.ClientConfig) => Pulsar.Client;
+
+export class Instrumentation extends InstrumentationBase<typeof Pulsar.Client> {
+  static readonly COMPONENT = 'pulsar';
+  static readonly COMMON_ATTRIBUTES = {
+    [SemanticAttributes.MESSAGING_SYSTEM]: 'pulsar',
+  };
+
+  constructor(config: InstrumentationConfig = {}) {
+    super(
+      '@opentelemetry/instrumentation-pulsar-client',
+      VERSION,
+      Object.assign({}, config)
+    );
+  }
+
+  override setConfig(config: InstrumentationConfig = {}) {
+    this._config = Object.assign({}, config);
+  }
```

Comment:
If you use the Instrumentation config, I don't think you need to override this method

Reply:
True, I removed and did some tests, everything looks fine without overriding this method